src/common/cockpitwebserver.c

```diff
@@ -837,16 +821,29 @@ cockpit_web_request_process (CockpitWebRequest *self,
     }
 
   self->original_path = path;
-  self->path = path + self->web_server->url_root->len;
   self->method = method;
   self->headers = headers;
 
+  g_autofree gchar *path_copy = g_strdup (self->original_path + self->web_server->url_root->len);
+  self->path = path_copy;
```

Comment:
Note to self: This looks dangerous/wrong at first, but it's just for calling the handler_stream signal handler, and it will be reset to NULL afterwards.